pkg/controller/erspan.go

```diff
@@ -0,0 +1,428 @@
+		podifdata := &EndPointData{
+			MacAddr:    podif.Status.MacAddr,
+			EPG:        deconEPG,
+			Namespace:  podif.Status.PodNS,
+			AppProfile: deconAppProfile,
+		}
+		podifKey := getPodifKey(podif)
+		cont.podIftoEp[podifKey] = podifdata
+	} else {
+		cont.log.Infof("Invalid EPG: %s", podif.Status.EPG)
+	}
+	cont.indexMutex.Unlock()
+}
+
+func (cont *AciController) podIFDeleted(obj interface{}) {
+	podif := obj.(*podIfpolicy.PodIF)
+	cont.log.Infof("podif Deleted: %s", podif.ObjectMeta.Name)
+	cont.indexMutex.Lock()
+	delete(cont.podIftoEp, podif.ObjectMeta.Name)
+	cont.indexMutex.Unlock()
+}
+
+func (cont *AciController) erspanPolicyUpdated(obj interface{}) {
+	erspanPolicy := obj.(*erspanpolicy.ErspanPolicy)
+	spankey, err := cache.MetaNamespaceKeyFunc(erspanPolicy)
+	if err != nil {
+		ErspanPolicyLogger(cont.log, erspanPolicy).
+			Error("Could not create key:" + err.Error())
+		return
+	}
+	cont.erspanPolPods.UpdateSelectorObj(obj)
+	cont.queueErspanUpdateByKey(spankey)
+}
+
+func (cont *AciController) queueErspanUpdateByKey(key string) {
+	cont.log.Debug("erspan key queued: ", key)
+	cont.erspanQueue.Add(key)
+}
+
+func (cont *AciController) erspanPolicyDeleted(obj interface{}) {
+	span, isSpan := obj.(*erspanpolicy.ErspanPolicy)
+	if !isSpan {
+		deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+		if !ok {
+			ErspanPolicyLogger(cont.log, span).
+				Error("Received unexpected object: ", obj)
+			return
+		}
+		span, ok = deletedState.Obj.(*erspanpolicy.ErspanPolicy)
+		if !ok {
+			ErspanPolicyLogger(cont.log, span).
+				Error("DeletedFinalStateUnknown contained non-erspan object: ", deletedState.Obj)
+			return
+		}
+	}
+	spankey, err := cache.MetaNamespaceKeyFunc(span)
+	if err != nil {
+		ErspanPolicyLogger(cont.log, span).
+			Error("Could not create erspan key: ", err)
+		return
+	}
+	cont.apicConn.ClearApicObjects(cont.aciNameForKey("span", spankey))
+	cont.erspanPolPods.DeleteSelectorObj(obj)
+
+}
+
+func (cont *AciController) erspanSyncOpflexDev() {
+	cache.ListAll(cont.erspanIndexer, labels.Everything(),
+		func(spanObj interface{}) {
+			cont.queueErspanUpdate(spanObj.(*erspanpolicy.ErspanPolicy))
+		})
+}
+
+func (cont *AciController) queueErspanUpdate(span *erspanpolicy.ErspanPolicy) {
+	key, err := cache.MetaNamespaceKeyFunc(span)
+	if err != nil {
+		ErspanPolicyLogger(cont.log, span).
+			Error("Could not create span key:" + err.Error())
+		return
+	}
+	cont.erspanQueue.Add(key)
+}
+
+func (cont *AciController) initErspanPolPodIndex() {
+	cont.erspanPolPods = index.NewPodSelectorIndex(cont.log,
+		cont.podIndexer, cont.namespaceIndexer, cont.erspanIndexer,
+		cache.MetaNamespaceKeyFunc,
+		func(obj interface{}) []index.PodSelector {
+			span := obj.(*erspanpolicy.ErspanPolicy)
+			ls := &metav1.LabelSelector{MatchLabels: span.Spec.Selector.Labels}
+			return index.PodSelectorFromNsAndSelector(span.Spec.Selector.Namespace, ls)
+		},
+	)
+	cont.erspanPolPods.SetPodUpdateCallback(func(podkey string) {
```

Comment:
do we need this change?  calling pod update  we are not doing anything specific to ERSPAN, may 
be we can remove this code

Reply:
Yes, you are right, will clean this up, thanks